compiler/rustc_mir_transform/src/deref_separator.rs

```diff
@@ -0,0 +1,63 @@
+use crate::MirPass;
+use rustc_middle::mir::patch::MirPatch;
+use rustc_middle::mir::*;
+use rustc_middle::ty::TyCtxt;
+pub struct Derefer;
+
+pub fn deref_finder<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+    let mut patch = MirPatch::new(body);
+    let (basic_blocks, local_decl) = body.basic_blocks_and_local_decls_mut();
+    for (block, data) in basic_blocks.iter_enumerated_mut() {
+        for (i, stmt) in data.statements.iter_mut().enumerate() {
+            match stmt.kind {
+                StatementKind::Assign(box (og_place, Rvalue::Ref(region, borrow_knd, place))) => {
+                    if borrow_knd == (BorrowKind::Mut { allow_two_phase_borrow: false }) {
```

Comment:
I think we could do this for all borrows, not just mutable ones.

Reply:
Was getting ICE while building the core without this now I deleted it and it's gone 🤷‍♂️

Reply:
oh, nevermind got it again 🤦‍♂️